packages/expo-updates/android/src/main/java/expo/modules/updates/launcher/DatabaseLauncher.kt

```diff
@@ -204,9 +204,12 @@ class DatabaseLauncher(
     return if (!assetFileExists) {
       // we still don't have the asset locally, so try downloading it remotely
       assetsToDownload++
+
+      val extraHeaders = FileDownloader.getExtraHeadersForRemoteAssetRequest(launchedUpdate, embeddedUpdate?.updateEntity, requestedUpdate)
```

Comment:
please double check if the callsite doesn't recompute extra headers for each assets. since the method is inside `fun ensureAssetExists(asset: AssetEntity, ...)`. i feel it's still a processor for "each" asset.